src/routes/fest-route/fest-discussion.jsx

```diff
@@ -0,0 +1,102 @@
+import { CREATE_DISCUSSION, GET_DISCUSSIONS } from "../../graphql/index.js";
+import { CurrentUserContext } from "../../store/current-user-context.js";
+
+export const FestDiscussion = ({ discussionQuery, festQuery, festId }) => {
+  const [discussionContent, setDiscussionContent] = useState("");
+  const { currentUser } = useContext(CurrentUserContext);
+  const userId = currentUser?.id;
+  const isValid = discussionContent.length !== 0 ? true : false;
+
+  // Create an array of all attendees to later check if the current user is in list
+  const attendeesList = festQuery?.data?.fest?.attendees?.map(
+    (person) => person?.id
+  );
+
+  // Mutation to add discussion to server, and refetch the queries
+  const [
+    createDiscussion,
+    { discussionData, discussionIsLoading, discussionHasError },
+  ] = useMutation(CREATE_DISCUSSION, {
+    refetchQueries: [GET_DISCUSSIONS],
+  });
+
+  // Function to handle clicking of 'send' button; sends API response
+  const handleDiscussionSubmit = (evt) => {
+    evt.preventDefault();
+    createDiscussion({
+      variables: {
+        data: {
+          content: discussionContent,
+          user: { connect: { id: userId } },
+          fest: { connect: { id: festId } },
+        },
+      },
+    });
+    if (discussionIsLoading) return "Submitting...";
+    else if (discussionHasError) return `Submission error! ${error.message}`;
+    else setDiscussionContent(""); // reset discussionContent to blank
+  };
+
+  return (
+    <div className="w-3/5">
+      <div>
+        {/* If discussionQuery is loading or the length is 0, show "no notes here yet" */}
+        {!discussionQuery?.loading &&
+        discussionQuery &&
+        discussionQuery?.data?.fest?.festNotes?.length === 0 ? (
+          <h2 className="font-arial text-lg/4 m-auto text-center pb-96">
+            No notes here yet
+          </h2>
+        ) : (
+          // otherwise show the posts coming from the backend
+          <div className="h-96 overflow-y-scroll">
+            {discussionQuery?.data?.fest?.festNotes
+              .map((discussion) => (
+                // Map each item onto the DiscussionCard template
+                <DiscussionCard
+                  discussion={discussion}
+                  key={discussion?.id ?? discussion._id}
+                />
+              ))
+              .sort((a, b) => {
+                // sort posts based on their time elapsed in milliseconds since 1970, placing more recent discussions at bottom
+                return (
+                  Date.parse(new Date(a.props.discussion.createdAt)) -
+                  Date.parse(new Date(b.props.discussion.createdAt))
+                );
+              })}
+          </div>
+        )}
+      </div>
+      {/* Check for the userId is in attendees list (only attendees can write comments); if not hide option to write discussion */}
+      {attendeesList?.includes(userId) && (
+        <div className="flex gap-x-10 mt-6">
+          <textarea
```

Comment:
Discussion text area stays the same height and adds a scroll bar when text wraps to new line.

Reply:
No changes needed right?

Reply:
Last week, I asked josh if I should make the text area grow in size and he said, we should leave it to start as one line and allow for the user to expand up to a certain height. It starts off as a height of 44 px, as suggested by figma, and the user can grow it to 80px

Reply:
Okay then it should be fine